lib/model/store.dart

```diff
@@ -786,9 +789,12 @@ class UpdateMachine {
           queueId: queueId, lastEventId: lastEventId);
       } catch (e) {
         store.isLoading = true;
+        final localizations = GlobalLocalizations.zulipLocalizations;
+        final serverUrl = store.connection.realmUrl.origin;
         switch (e) {
           case ZulipApiException(code: 'BAD_EVENT_QUEUE_ID'):
             assert(debugLog('Lost event queue for $store.  Replacing…'));
+            reportErrorToUser(localizations.errorReconnectingToServer(serverUrl));
             await store._globalStore._reloadPerAccount(store.accountId);
             dispose();
```

Comment:
How about hiding the snackbar as soon as we have a new event queue?

```suggestion
            reportErrorToUser(localizations.errorReconnectingToServer(serverUrl));
            await store._globalStore._reloadPerAccount(store.accountId);
            ZulipApp.scaffoldMessenger?.hideCurrentSnackBar();
            dispose();
```

(Or maybe `reportErrorToUser` can return a callback, or return the `ScaffoldFeatureController` returned by `ScaffoldMessengerState.showSnackBar`, that we can use to hide the snackbar if one was shown.)

In my manual testing, the reconnection to CZO is finished before the snackbar disappears, but the snackbar is still present and holding my attention for some time, making it seem (I think) like the reconnection is taking longer than it actually takes. Also, it looks like maybe `ScaffoldMessengerState.showSnackBar` lets us make the exit animation duration shorter than the entrance animation, with its `snackBarAnimationStyle` param, and that might also be something we could do to help make the app feel snappy and responsive.

Reply:
These all sound like great ideas, thanks.